File: tasks/3945-digit-frequency-score.go
// Accepted 1000 / 1000 testcases passed Sergey Pomortsev submitted at Jul 02, 2026 15:44
//  Solution Runtime 0 ms Beats 100.00% Memory 4.10 MB Beats 9.71%

import "strconv"

func digitFrequencyScore(n int) int {
  var freq = make(map[rune]int)
  var str = strconv.Itoa(n)
  var result int

  for _, symb := range str {
    freq[symb]++
  }

  for r, num := range freq {
    result += num * int(r - '0')
  }

  return result
}

File: tasks/3945-digit-frequency-score.ts
// Accepted 1000 / 1000 testcases passed Sergey Pomortsev submitted at Jun 30, 2026 18:20
// Solution Runtime 1 ms Beats 78.53% Memory 58.64 MB Beats 14.14%

function digitFrequencyScore(n: number): number {
  const arr = new Array(10).fill(0);
  const str = String(n);

  for (let i = 0; i < str.length; i++) {
    arr[Number(str[i])]++;
  }

  return arr.reduce((acc, curr, i) => acc + curr * i, 0);
};